cdap-securestore-spi/src/main/java/co/cask/cdap/securestore/spi/SecretStore.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+package co.cask.cdap.securestore.spi;
+
+import co.cask.cdap.securestore.spi.secret.Decoder;
+import co.cask.cdap.securestore.spi.secret.Encoder;
+
+import java.util.Collection;
+
+/**
+ * Stores and retrieves secrets and associated metadata for the secrets. It must not be used to store
+ * secrets/sensitive information in plain text.
+ *
+ */
+public interface SecretStore {
+
+  /**
+   * Retrieves stored data for the secret.
+   *
+   * @param namespace the namespace to which secret belongs to
+   * @param name the name of the secret to be retrieved
+   * @return stored data
+   * @throws SecretNotFoundException if secret is not found in provided namespace
+   * @throws Exception if unable to retrieve the secret
```

Comment:
Similar to comment above. I think the exception should be more specific, like `IOException`.

Reply:
done